### PR TITLE
Treat line numbers as proper marks

### PR DESCRIPTION
**Pull request by AndreasArvidsson**

### CODE REVIEW

File: src/test/suite/fixtures/recorded/lineNumbers/takeRowFour.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ command:
   partialTargets:
     - type: primitive
       selectionType: line
-      modifier:
```
Comment (pokey) [inline src/test/suite/fixtures/recorded/lineNumbers/takeRowFour.yml:8]: shall we add a test case that uses it like a proper mark?  Eg "take funk row five"

File: src/test/suite/fixtures/recorded/lineNumbers/takeRowFour.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@ command:
   partialTargets:
     - type: primitive
       selectionType: line
```
Comment (pokey) [inline src/test/suite/fixtures/recorded/lineNumbers/takeRowFour.yml:7]: So this `selectionType` comes from talon side?  Probably ok but came as a bit of a surprise.  I wonder if we could infer that info somewhere in the extension, either during inference or processing

**Review by pokey — COMMENTED** [review]
LGTM with a couple minor comments

File: src/test/suite/fixtures/recorded/lineNumbers/takeRowFour.yml (hunk 1 above)
Comment (AndreasArvidsson) [reply src/test/suite/fixtures/recorded/lineNumbers/takeRowFour.yml:8]: We don't really have that for the other marks. There is no "take funk that" or similar. But it is treated like the other marks in the code so there really is no difference. If we should add more test for marks in general is a different question.

File: src/test/suite/fixtures/recorded/lineNumbers/takeRowFour.yml (hunk 2 above)
Comment (AndreasArvidsson) [reply src/test/suite/fixtures/recorded/lineNumbers/takeRowFour.yml:7]: We could enter it extension side if we want. But that is true for several other commands as well. This solution has the upside that we don't add additional code if I don't actually need to. Instead we reuse the functionality of the line selector and the mark actually only selects the first character of the correct line.

File: src/test/suite/fixtures/recorded/lineNumbers/takeRowFour.yml (hunk 1 above)
Comment (pokey) [reply src/test/suite/fixtures/recorded/lineNumbers/takeRowFour.yml:8]: I think I'd like to have that kind of test, but if you think it's better to do those in one pass that's fine too; we don't need to block this PR on that

Comment (AndreasArvidsson) [reply src/test/suite/fixtures/recorded/lineNumbers/takeRowFour.yml:8]: Exactly I think that we probably should do proper tests for all the marks as a separate thing and not just this one specifically.

**Review by pokey — APPROVED** [review]

**Pull request merged**